plugins/source/mixpanel/resources/services/export/export_events.go

```diff
@@ -103,11 +108,18 @@ func resolveExportTime(_ context.Context, meta schema.ClientMeta, r *schema.Reso
 	return r.Set(c.Name, time.Unix(int64(tf), 0))
 }
 
-func resolveDistinctID(_ context.Context, meta schema.ClientMeta, r *schema.Resource, c schema.Column) error {
+// resolveDistinctID will try "distinct_id" & "$insert_id" properties first, generating some random ID if unsuccessful.
+func resolveDistinctID(_ context.Context, _ schema.ClientMeta, r *schema.Resource, c schema.Column) error {
 	e := r.Item.(mixpanel.ExportEvent)
-	val, ok := e.Properties["distinct_id"]
-	if !ok {
-		return errors.New("event does not have a distinct_id property")
+
+	var val any
+	if v, ok := e.Properties[distinctID]; ok {
+		val = v
+	} else if v, ok = e.Properties[insertID]; ok {
+		val = v
```

Comment:
IMO we need some ID there. It's still in the properties, too.

I'm OK with just random one, too

Reply:
Using `$insertId` is fine as it's there to deduplicate events in case they get duplicated during a "at least once" transfer. My point was about the name of the column. Maybe just call it `id`? Would be breaking though.

Reply:
🤷‍♂️

Reply:
I think if we cannot guarantee uniqueness we should just drop the PK...

Reply:
we still need to it to not have the duplicates on the source side, so that's why I'd rather use this kludge atm and fix it properly if anyone asks and provides more insight.

Reply:
This doesn't stop us from having duplicates anymore than having no PK does

Reply:
I'll let @disq decide as he did the initial implementation

Reply:
it's slightly better than having no PKs and constantly duplicated data, as as far as we can tell missing the distinct_id could be a rare occurance, and docs say $insert_id should be there. So I doubt we'll fall into the random id case often, but that's just a guess.